### PR TITLE
Feat/interverse

### PR DESCRIPTION
**Pull request by stnw**
**WHAT?**
Functionality for the cloud-ui to reach the interverse API on charles-client-accounts-api.
The enpoints: https://github.com/c-commerce/charles-browser-sdk/pull/122/files#diff-122bda192034b3469ed452a2e55213c140de3003defc3d0245d47b5686f363d2R361

### CODE REVIEW

File: src/cloud/index.ts [hunk 1]
```diff
@@ -332,7 +336,8 @@ export class Cloud extends APICarrier {
 
   public async makeBaseResourceListRequest<T, TL, K, O, E>(proto: BaseResourceCreateable<T, K>, listProto: BaseResourceList<TL>, errorProto: BaseResourceErrorProto<E>, options?: BaseResourceEntityFetchOptions<O>): Promise<T[] | K[] | undefined> {
     try {
-      const res = await this.http.getClient().get(`${this.cloudBase}/${listProto.endpoint}`, {
+      const endpoint = options?.endpoint ?? listProto.endpoint
+      const res = await this.http.getClient().get(`${this.cloudBase}/${endpoint}`, {
```
Comment (stnw) [inline src/cloud/index.ts:340]: In the current implementation, it's only possible to have a static endpoint. I'm adding the possibility to add a dynamic endpoint i.g. `api/v0/interverse/organizations/${this.organizationId}/keys`

**Review by haayhappen — APPROVED** [review]
LGTM

File: src/cloud/index.ts [hunk 2]
```diff
@@ -352,6 +357,27 @@ export class Cloud extends APICarrier {
...
   }
 
+  public interverse = {
+    organizations: async (options?: EntityFetchOptions): Promise<interverseOrganization.InterverseOrganization[] | interverseOrganization.InterverseOrganizationRawPayload[] | undefined> => {
+      return await this.makeBaseResourceListRequest<interverseOrganization.InterverseOrganization, interverseOrganization.InterverseOrganizations, interverseOrganization.InterverseOrganizationRawPayload, EntityFetchOptions, interverseOrganization.InterverseOrganizationsFetchRemoteError>(interverseOrganization.InterverseOrganization, interverseOrganization.InterverseOrganizations, interverseOrganization.InterverseOrganizationsFetchRemoteError, options)
+    },
+    organisation: (payload: interverseOrganization.InterverseOrganizationRawPayload): interverseOrganization.InterverseOrganization => {
+      return interverseOrganization.InterverseOrganization.create(payload, this, this.http)
+    },
+    apiKeys: async (options?: EntityFetchOptions): Promise<apiKey.ApiKey[] | apiKey.ApiKeyRawPayload[] | undefined> => {
```
Comment (artdnlv) [inline src/cloud/index.ts:367]: This `apiKey.ApiKey[] | apiKey.ApiKeyRawPayload[]` has given me some pain recently. When typ checking is enabled on agent UI, I have no way to know which one on the two is being returned. Here is what I came up with: 

* https://github.com/c-commerce/charles-browser-sdk/blob/master/src/universe/index.ts#L147-L150
* https://github.com/c-commerce/charles-browser-sdk/blob/master/src/universe/index.ts#L308

**Review by artdnlv — APPROVED** [review]
LGTM

Added a few sidenotes/FYI.

**Comment by haayhappen** [comment]
Merging to be able to test it without locally integrating the sdk

**Pull request merged**